packages/extension/src/ui/features/multisig/MultisigConfirmationsScreen.tsx

```diff
@@ -1,37 +1,142 @@
 import { H1, H4, P3 } from "@argent/ui"
-import { Box, Button, Center } from "@chakra-ui/react"
+import { Box, Button, Center, Flex } from "@chakra-ui/react"
 import { FC } from "react"
-import { useFormContext } from "react-hook-form"
+import { FormProvider, useFormContext } from "react-hook-form"
 
+import {
+  addMultisigOwners,
+  updateMultisigThreshold,
+} from "../../../shared/multisig/multisig.service"
 import { Account } from "../accounts/Account"
 import { useRouteAccount } from "../shield/useRouteAccount"
-import { FieldValues } from "./hooks/useCreateMultisigForm"
+import { FieldValuesCreateMultisigForm } from "./hooks/useCreateMultisigForm"
 import { useMultisigInfo } from "./hooks/useMultisigInfo"
+import {
+  FieldValuesThresholdForm,
+  useUpdateThresholdForm,
+} from "./hooks/useUpdateThreshold"
 import { MultisigSettingsWrapper } from "./MultisigSettingsWrapper"
 import { SetConfirmationsInput } from "./SetConfirmationsInput"
 
 export const MultisigConfirmationsScreen: FC = () => {
   const account = useRouteAccount()
   return (
     <MultisigSettingsWrapper>
-      {account && <MultisigConfirmations account={account} />}
+      {account && <MultisigConfirmationsWithFormProvider account={account} />}
     </MultisigSettingsWrapper>
   )
 }
 
-export const MultisigConfirmations = ({ account }: { account: Account }) => {
+const MultisigConfirmationsWithFormProvider = ({
+  account,
+}: {
+  account: Account
+}) => {
   const { multisig } = useMultisigInfo(account)
-  const { trigger } = useFormContext<FieldValues>()
+
+  const methods = useUpdateThresholdForm(multisig?.threshold)
+  return (
+    <FormProvider {...methods}>
+      <MultisigConfirmationsWithoutOwners account={account} />
+    </FormProvider>
+  )
+}
+export const MultisigConfirmationsWithOwners = ({
+  account,
+}: {
+  account: Account
+}) => {
+  const { multisig } = useMultisigInfo(account)
+  const {
+    trigger,
+    formState: { errors },
+    getValues,
+  } = useFormContext<FieldValuesCreateMultisigForm>()
+
   const handleNextClick = () => {
     trigger()
-    // TODO: implement
+    if (!Object.keys(errors).length) {
```

Comment:
would probably use lodash-es `isEmpty`, feels more clean
```suggestion
    if (isEmpty(errors)) {
```

Reply:
thanks !